Formula/w/whisper-cpp.rb

```diff
@@ -33,11 +33,27 @@ def install
       -DWHISPER_BUILD_SERVER=OFF
     ]
     args << "-DLLAMA_METAL_MACOSX_VERSION_MIN=#{MacOS.version}" if OS.mac?
+    args << "-DCMAKE_INSTALL_RPATH=#{OS.mac? ? "@loader_path/../libinternal" : "'$ORIGIN/../libinternal'"}" if head?
 
-    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
-    # the "main" target is our whisper-cpp binary
-    system "cmake", "--build", "build", "--target", "main"
-    bin.install "build/bin/main" => "whisper-cpp"
+    # avoid installing libggml libraries to "lib" since they would conflict with llama.cpp
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args(install_libdir: "libinternal")
```

Comment:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args(install_libdir: "lib/ggml")
```
Maybe? You'll need to adjust the argument to `#rpath` from my suggestion above if you apply this. (to `rpath(target: lib/"ggml")`).

Though, ideally we should be using `libggml` from `llama.cpp` instead.

Reply:
> Though, ideally we should be using `libggml` from `llama.cpp` instead (or vice-versa).

I've suggested this with maintainers of libggml/llama.cpp/whisper.cpp and they said that could be an eventual goal but that the codebases aren't ready for that at this moment since libggml seems to be such a fast-moving target.

Reply:
If we're going to be installing into a private `libinternal` directory anyway: why not do `BUILD_SHARED_LIBS=OFF` instead? That will avoid the need to specify a special `install_libdir` and a `CMAKE_INSTALL_RPATH` flag.

Reply:
I've asked that too of the maintainers and they said that building with shared libraries (even if no other program is using those libraries) is preferred for optimizations at runtime because the binary can load the correct library for currently available CPUs or GPUs. So, it sounds like it's an optimization thing.